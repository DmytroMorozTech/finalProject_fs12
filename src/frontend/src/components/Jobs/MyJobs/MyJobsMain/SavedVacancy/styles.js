import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({

  vacancy: {
    display: 'flex',
    paddingTop: theme.spacing(3),
    paddingBottom: theme.spacing(2),
    justifyContent: 'space-between',
    width: '100%'
  },

  row: {
    display: 'flex'
  },

  vacancyImg: {
    width: theme.spacing(10),
    marginLeft: theme.spacing(3),
    marginRight: theme.spacing(3)
  },

  link: {
    display: 'flex',
    fontSize: theme.typography.h5.fontSize,
    fontWeight: theme.typography.fontWeightBold,
    color: theme.palette.grey[900],
    cursor: 'pointer',
    '&:hover': {
      color: theme.palette.primary.main,
      textDecoration: 'underline',
      textDecorationThickness: '2px'
    }
  },

  org: {
    color: theme.palette.grey[900]
  },

  recruiting: {
    display: 'flex',
    marginTop: theme.spacing(1),
    marginBottom: theme.spacing(1),
    alignItems: 'center'
  },

  iconTarget: {
    color: theme.palette.success.main,
    fontSize: theme.typography.icons.medium.fontSize,
    marginRight: theme.spacing(1)
  },

  date: {
    display: 'flex',
    marginTop: theme.spacing(2)
  },

  applicants: {
    color: theme.palette.success.main,
    fontWeight: theme.typography.fontWeightBold
  },

  popupMenu: {
    marginRight: theme.spacing(5),
    marginTop: theme.spacing(2)
  },

  line: {
    width: '100%',
    height: '0.5px',
    border: '0',
    margin: '0 auto',
    backgroundColor: theme.palette.grey[300]
  }
}))
