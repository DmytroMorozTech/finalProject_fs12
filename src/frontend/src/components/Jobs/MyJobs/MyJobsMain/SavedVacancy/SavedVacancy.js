import styles from './styles'
import SmallDot from '../../../../../shared/SmallDot/SmallDot'
import Typography from '@material-ui/core/Typography'
import React from 'react'
import TrackChangesIcon from '@material-ui/icons/TrackChanges'
import SimpleMenu from '../../../../../shared/PopupMenu/PopupMenu'
import ThreeDots from '../../../../../shared/ThreeDots/TreeDots'
import SavedVacancyAdditions from './SavedVacancyAdditions/SavedVacancyAdditions'
import Image from '../../../../../shared/Image/Image'

function SavedVacancy (props) {
  const {
    titleOfVacancy = 'Junior Java Developer',
    organizationName = 'GlobalLogic',
    organizationLocation = 'Kyiv, Ukraine',
    createdDate = '3 weeks ago',
    applicants = 2
  } = props

  const classes = styles()

  return (
    <div>
      <div>
        <div className={classes.vacancy}>
          <div className={classes.row}>
            <div>
              <Image
                imageUrl={'linkedin/general/mr5jb7fifohjgy8d7unj'}
                className={classes.vacancyImg}
                alt={'job'}
              />
            </div>
            <div>
              <div className={classes.link}>
                {titleOfVacancy}
              </div>
              <Typography variant="h6" className={classes.org}>
                {organizationName}
              </Typography>
              <Typography variant="h6">
                {organizationLocation}
              </Typography>
              <div className={classes.recruiting}>
                <TrackChangesIcon fontSize="inherit" className={classes.iconTarget}/>
                <Typography variant="h6">
                  Actively recruiting
                </Typography>
              </div>
              <div className={classes.date}>
                <Typography variant="h6">
                  {createdDate}
                </Typography>
                <SmallDot/>
                <Typography variant="h6" className={classes.applicants}>
                  {applicants} applicants
                </Typography>
              </div>
            </div>
          </div>
          <div className={classes.popupMenu}>
            <SimpleMenu menuItem={
              <ThreeDots/>
            } userData={<SavedVacancyAdditions/>}/>
          </div>
        </div>
      </div>
      <div>
        <hr className={classes.line}/>
      </div>
    </div>
  )
}

export default SavedVacancy
