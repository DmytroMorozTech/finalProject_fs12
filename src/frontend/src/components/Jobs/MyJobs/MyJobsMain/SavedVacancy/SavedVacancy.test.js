import { shallow } from 'enzyme'
import React from 'react'
import SavedVacancy from './SavedVacancy'

jest.mock('./styles', () => () => ({}))
jest.mock('react-router-dom', () => ({
  Link: 'a'
}))

const setUp = () => shallow(<SavedVacancy/>)

describe('should render SavedVacancy component', () => {
  let component
  beforeEach(() => {
    component = setUp()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
