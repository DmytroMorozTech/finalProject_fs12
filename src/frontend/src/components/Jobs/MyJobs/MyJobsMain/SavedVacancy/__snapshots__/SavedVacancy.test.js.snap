// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SavedVacancy component to match snapshot 1`] = `
<div>
  <div>
    <div>
      <div>
        <div>
          <Image
            alt="job"
            imageUrl="linkedin/general/mr5jb7fifohjgy8d7unj"
          />
        </div>
        <div>
          <div>
            Junior Java Developer
          </div>
          <WithStyles(ForwardRef(Typography))
            variant="h6"
          >
            GlobalLogic
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            variant="h6"
          >
            Kyiv, Ukraine
          </WithStyles(ForwardRef(Typography))>
          <div>
            <Memo(ForwardRef(TrackChangesIcon))
              fontSize="inherit"
            />
            <WithStyles(ForwardRef(Typography))
              variant="h6"
            >
              Actively recruiting
            </WithStyles(ForwardRef(Typography))>
          </div>
          <div>
            <WithStyles(ForwardRef(Typography))
              variant="h6"
            >
              3 weeks ago
            </WithStyles(ForwardRef(Typography))>
            <SmallDot />
            <WithStyles(ForwardRef(Typography))
              variant="h6"
            >
              2
               applicants
            </WithStyles(ForwardRef(Typography))>
          </div>
        </div>
      </div>
      <div>
        <SimpleMenu
          menuItem={<ThreeDots />}
          userData={<SavedVacancyAdditions />}
        />
      </div>
    </div>
  </div>
  <div>
    <hr />
  </div>
</div>
`;
